src/edu/csus/ecs/pc2/core/Utilities.java

```diff
@@ -960,58 +976,75 @@ public long stringToLongSecs(String s) {
     /**
      * Locate judges data file on disk.
      * 
+     * @see Utilities#locateJudgesDataFile(Problem, String, String, DataFileType)
      * @param problem
      * @param serializedFile
+     * @param alternateCDPPath
      * @param judgeDataFile
-     * @return
      */
     public static String locateJudgesDataFile(Problem problem, SerializedFile serializedFile, String alternateCDPPath, DataFileType judgeDataFile) {
 
-        if (serializedFile.isExternalFile()) {
-
-            String testFileName;
-
-            if (alternateCDPPath != null && alternateCDPPath.trim().length() > 0) {
-
-                testFileName = getSecretDataPath(alternateCDPPath, problem) + File.separator + serializedFile.getName();
-                if (fileExists(testFileName)) {
-                    return testFileName;
-                }
-            }
+        String testFileName = locateJudgesDataFile(problem, serializedFile.getName(), alternateCDPPath);
+        
+        if (testFileName != null && fileExists(testFileName)) {
+            return testFileName;
+        }
+        
+        testFileName = serializedFile.getAbsolutePath();
 
-            String secretPathPattern = File.separator + SECRET_DATA_DIR + File.separator;
-            String fullPathName = serializedFile.getAbsolutePath();
+        if (fileExists(testFileName)) {
+            return testFileName;
+        }
 
-            secretPathPattern = secretPathPattern.replace('\\', '.');
+        return null;
+    }
+    
+    /**
+     * find judges data file under secret/ or samples/.
+     * 
+     * @param problem
+     * @param baseFileName
+     * @param alternateCDPPath
+     * @param judgeDataFile
```

Comment:
This attribute describes an input parameter that doesn't exist in the method declaration.